apps/account-api/src/services/delegation.service.ts

```diff
@@ -47,26 +48,29 @@ export class DelegationService {
   }
 
   async getRevokeDelegationPayload(accountId: string, providerId: string): Promise<RevokeDelegationPayloadResponseDto> {
-    try {
-      const msaId = await this.blockchainService.publicKeyToMsaId(accountId);
-      if (!msaId) {
-        throw new HttpException('MSA ID for account not found', HttpStatus.NOT_FOUND);
-      }
+    if (!isAddress(accountId)) {
+      throw new HttpException('Invalid accountId', HttpStatus.BAD_REQUEST);
+    }
+    const msaId = await this.blockchainService.publicKeyToMsaId(accountId);
+    if (!msaId) {
+      throw new HttpException('MSA ID for account not found', HttpStatus.NOT_FOUND);
+    }
 
-      // Validate that the providerId is a registered provider, also checks for valid MSA ID
-      const providerRegistry = await this.blockchainService.getProviderToRegistryEntry(providerId);
-      if (!providerRegistry) {
-        throw new HttpException('Provider not found', HttpStatus.BAD_REQUEST);
-      }
+    // Validate that the providerId is a registered provider, also checks for valid MSA ID
+    const providerRegistry = await this.blockchainService.getProviderToRegistryEntry(providerId);
+    if (!providerRegistry) {
+      throw new HttpException('Provider not found', HttpStatus.BAD_REQUEST);
+    }
 
-      // Validate that delegations exist for this msaId
+    // Validate that delegations exist for this msaId
+    try {
```

Comment:
2 things here:
1. If `getDelegation()` returns anything at all, it will be a delegation for the current provider. So checking the providerId is unnecessary.
2. A delegation object would still be returned if the delegation was already revoked; you need to check `revokedAt` and compare to the current block number. (Actually, just that it's non-zero is fine; we only care about which block it was revoked at if we're checking for a delegation a some point in time past)